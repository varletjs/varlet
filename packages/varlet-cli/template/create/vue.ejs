<template>
  <div class="var-<%- kebabCaseName %>"></div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import './<%- kebabCaseName %>.less'

export default defineComponent({
  name: '<%- namespace + bigCamelizeName %>',
})
</script>
