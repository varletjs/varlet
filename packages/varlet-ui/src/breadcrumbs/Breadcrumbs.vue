<template>
  <div :class="classes(n())">
    <slot />
  </div>
</template>

<script lang="ts">
import { defineComponent, computed } from 'vue'
import { props } from './props'
import { useBreadcrumbsList } from './provide'
import { createNamespace } from '../utils/components'
import type { BreadcrumbsProvider } from './provide'
import type { ComputedRef } from 'vue'

const { n, classes } = createNamespace('breadcrumbs')

export default defineComponent({
  name: 'VarBreadcrumbs',
  inheritAttrs: false,
  props,
  setup(props) {
    const separator: ComputedRef<string> = computed(() => props.separator)
    const { bindBreadcrumbList, length } = useBreadcrumbsList()

    const breadcrumbsProvide: BreadcrumbsProvider = {
      length,
      separator,
    }

    bindBreadcrumbList(breadcrumbsProvide)

    return {
      n,
      classes,
    }
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import './breadcrumbs';
</style>
