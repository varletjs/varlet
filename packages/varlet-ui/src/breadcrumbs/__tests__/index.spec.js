import Breadcrumbs from '..'
import { createApp } from 'vue'
import { mount } from '@vue/test-utils'

test('test breadcrumb plugin', () => {
  const app = createApp({}).use(Breadcrumbs)
  expect(app.component(Breadcrumbs.name)).toBeTruthy()
})
