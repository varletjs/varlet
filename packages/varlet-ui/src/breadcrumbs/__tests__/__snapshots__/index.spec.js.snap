// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test breadcrumbs basic usage 1`] = `
"<div class="var-breadcrumbs">
  <div class="var-breadcrumb">
    <div class="var-breadcrumb__content var-breadcrumb--active">HOME</div>
    <div class="var-breadcrumb__separator">/</div>
  </div>
  <div class="var-breadcrumb">
    <div class="var-breadcrumb__content var-breadcrumb--active">LINK 1</div>
    <div class="var-breadcrumb__separator">/</div>
  </div>
  <div class="var-breadcrumb">
    <div class="var-breadcrumb__content">LINK 2</div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`test breadcrumbs parent and child separator 1`] = `
"<div class="var-breadcrumbs">
  <div class="var-breadcrumb">
    <div class="var-breadcrumb__content var-breadcrumb--active">HOME</div>
    <div class="var-breadcrumb__separator">-</div>
  </div>
  <div class="var-breadcrumb">
    <div class="var-breadcrumb__content var-breadcrumb--active">LINK 1</div>
    <div class="var-breadcrumb__separator">-</div>
  </div>
  <div class="var-breadcrumb">
    <div class="var-breadcrumb__content">LINK 2</div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`test breadcrumbs parent separator 1`] = `
"<div class="var-breadcrumbs">
  <div class="var-breadcrumb">
    <div class="var-breadcrumb__content var-breadcrumb--active">HOME</div>
    <div class="var-breadcrumb__separator">+</div>
  </div>
  <div class="var-breadcrumb">
    <div class="var-breadcrumb__content var-breadcrumb--active">LINK 1</div>
    <div class="var-breadcrumb__separator">+</div>
  </div>
  <div class="var-breadcrumb">
    <div class="var-breadcrumb__content">LINK 2</div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`test breadcrumbs slots 1`] = `
"<div class="var-breadcrumbs">
  <div class="var-breadcrumb">
    <div class="var-breadcrumb__content var-breadcrumb--active">HOME</div><span>*</span>
  </div>
  <div class="var-breadcrumb">
    <div class="var-breadcrumb__content var-breadcrumb--active">LINK 1</div><span>*</span>
  </div>
  <div class="var-breadcrumb">
    <div class="var-breadcrumb__content">LINK 2</div>
    <!--v-if-->
  </div>
</div>"
`;
